packages/framework-core/src/booster-graphql-dispatcher.ts

```diff
@@ -150,13 +156,15 @@ export class BoosterGraphQLDispatcher {
     return result
   }
 
-  private async handleDisconnect(envelope: GraphQLRequestEnvelope): Promise<void> {
-    if (!envelope.connectionID) {
+  private async handleDisconnect(connectionID?: string): Promise<void> {
+    if (!connectionID) {
       // This should be impossible, but just in case
       this.logger.debug("Received a DISCONNECT message but field 'connectionID' is missing. Doing nothing")
       return
     }
-    return this.readModelDispatcher.unsubscribeAll(envelope.connectionID)
+    this.logger.debug('Deleting all subscriptions and connection data')
+    await this.config.provider.connections.deleteData(this.config, connectionID)
+    await this.readModelDispatcher.unsubscribeAll(connectionID)
```

Comment:
Now, when the socket disconnects, we delete bot the subscriptions and the connection data